gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/SplitFileService.java

```diff
@@ -0,0 +1,162 @@
+    private ResourceLoader resourceLoader;
+
+    @Autowired
+    private AmazonS3 amazonS3;
+
+
+    @Autowired
+    public SplitFileService(TypeService typeService,
+                            ServiceSelector serviceSelector,
+                            ProjectAgentRepository projectAgentRepository,
+                            DocumentService documentService,
+                            ResourceLoader resourceLoader) {
+        this.typeService = typeService;
+        this.serviceSelector = serviceSelector;
+        this.projectAgentRepository = projectAgentRepository;
+        this.documentService = documentService;
+        this.resourceLoader = resourceLoader;
+    }
+
+
+    public List<DocumentInstanceSection> splitDocumentToSections(DocumentCriteria criteria, UUID projectId) throws GendoxException, IOException {
+
+        List<DocumentInstanceSection> sections = new ArrayList<>();
+        Page<DocumentInstance> documentInstances = documentService.getAllDocuments(criteria);
+
+
+        for (DocumentInstance documentInstance : documentInstances) {
+
+            // get file
+            InputStream inputStream = downloadFile(documentInstance.getRemoteUrl());
+
+            // files content
+            String content = readTxtFileContent(inputStream);
+
+            List<DocumentInstanceSection> documentSections = createSections(documentInstance, content, projectId);
+
+            for (DocumentInstanceSection section : documentSections) {
+                sections.add(section);
+            }
+
+        }
+        return sections;
+    }
+
+    private String readTxtFileContent(InputStream inputStream) throws IOException {
+        StringBuilder fileContent = new StringBuilder();
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                fileContent.append(line).append(" \n ");
+            }
+        }
+
+        return fileContent.toString();
+    }
+
+
+    public InputStream downloadFile(String fileUrl) throws GendoxException {
+
+        try {
+            Resource fileResource = resourceLoader.getResource(fileUrl);
+            InputStream inputStream = fileResource.getInputStream();
+            return inputStream;
+        } catch (Exception e) {
+            // Handle any exceptions
+            throw new GendoxException("ERROR_DOWNLOAD_FILE", "Error downloading file: " + fileUrl, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+
+    // Extract the file name from the URL (e.g., "http://example.com/path/to/file.pdf" -> "file.pdf")
+    private String extractFileNameFromUrl(String url) {
+        int lastSlashIndex = url.lastIndexOf("/");
+        if (lastSlashIndex >= 0) {
+            return url.substring(lastSlashIndex + 1);
+        } else {
+            return "downloaded-file"; // Default file name if extraction fails
+        }
```

Comment:
The file should have a name, I propose to through an exception instead of returning a default name.